File: python/rna-transcription/rna_transcription.py
def to_rna(a):
    d = {'C': 'G', 'G': 'C', 'T': 'A', 'A': 'U'}
    a = list(a)
    for i, l in enumerate(a):
        if l not in d.keys():
            return ''
        a[i] = d[a[i]]
    return ''.join(a)

File: elixir/rna-transcription/rna_transcription.exs
defmodule RNATranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RNATranscription.to_rna('ACTG')
  'UGAC'
  """
  @trans %{?A => ?U, ?C => ?G, ?T => ?A, ?G => ?C}

  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    dna |> Enum.map(&(@trans[&1]))
  end
end
